pkg/apis/postgres-operator.crunchydata.com/v1beta1/shared_types.go

```diff
@@ -5,10 +5,89 @@
 package v1beta1
 
 import (
+	"encoding/json"
+
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kube-openapi/pkg/validation/strfmt"
 )
 
+// ---
+// Duration represents a string accepted by the Kubernetes API in the "duration"
+// [format]. This format extends the "duration" [defined by OpenAPI] by allowing
+// some whitespace and more units:
+//
+//   - nanoseconds: ns, nano, nanos
+//   - microseconds: us, µs, micro, micros
+//   - milliseconds: ms, milli, millis
+//   - seconds: s, sec, secs
+//   - minutes: m, min, mins
+//   - hours: h, hr, hour, hours
+//   - days: d, day, days
+//   - weeks: w, wk, week, weeks
+//
+// An empty amount is represented as "0" with no unit.
+// One day is always 24 hours and one week is always 7 days (168 hours).
+//
+// +kubebuilder:validation:Format=duration
+// +kubebuilder:validation:MinLength=1
```

Comment:
Feel like I'm ⛏️ , but what's a MinLength of 1 for a Duration?

Reply:
It's a string, so number of characters.

Reply:
🤔 🤔 it's good to set a MaxLength, too, when there are [CEL] validation rules. I'll add that now.